src/api/spec/services/webui_controller_service/user_checker_spec.rb

```diff
@@ -2,67 +2,106 @@
   let(:user_checker) { described_class.new(http_request: request, config: config) }
 
-  describe '#proxy_enabled?' do
-    let(:user_login) { user.login }
-    let(:request) { OpenStruct.new(env: { 'HTTP_X_USERNAME' => user_login }) }
+  describe '#call' do
+    subject { user_checker.call }
 
-    subject { user_checker.proxy_enabled? }
-
-    context 'when its enabled' do
-      let(:config) { { 'proxy_auth_mode' => :on } }
-
-      it { expect(subject).to be(true) }
-    end
-
-    context 'when its disabled' do
+    context 'when the proxy authentication is disabled' do
+      let(:request) do
+        OpenStruct.new(env: { 'HTTP_X_USERNAME' => 'something',
+                              'HTTP_X_FIRSTNAME' => 'foo', 'HTTP_X_LASTNAME' => 'bar',
+                              'HTTP_X_EMAIL' => 'foo@example.org' })
+      end
       let(:config) { { 'proxy_auth_mode' => :off } }
 
-      it { expect(subject).to be(false) }
-    end
-  end
-
-  describe '#login_exists?' do
-    let(:request) { OpenStruct.new(env: { 'HTTP_X_USERNAME' => user_login }) }
-    let(:config) { { 'proxy_auth_mode' => :on } }
-
-    subject { user_checker.login_exists? }
-
-    context 'user exists' do
-      let(:user_login) { user.login }
+      before do
+        allow(User).to receive(:find_nobody!)
+      end
 
-      it { expect(subject).to be(true) }
-    end
-
-    context 'user does not exist' do
-      let(:user_login) { 'bar' }
-
-      it { expect(subject).to be(false) }
-    end
-  end
+      it { is_expected.to be(true) }
 
-  describe '#find_or_create_user!' do
-    let(:request) do
-      OpenStruct.new(env: { 'HTTP_X_USERNAME' => user_login,
-                            'HTTP_X_FIRSTNAME' => 'foo', 'HTTP_X_LASTNAME' => 'bar',
-                            'HTTP_X_EMAIL' => 'foo@example.org' })
+      it 'does not set the current user' do
+        subject
+        expect(User).not_to have_received(:find_nobody!)
+      end
     end
-    let(:config) { { 'proxy_auth_mode' => :on } }
 
-    subject { user_checker.find_or_create_user! }
-
-    context 'it will find an user' do
-      let(:user_login) { user.login }
-
-      it { expect(subject).to eq(user) }
-    end
-
-    context 'it will create an user' do
-      let(:user_login) { 'bar' }
+    context 'when the proxy authentication is enabled' do
+      let(:config) { { 'proxy_auth_mode' => :on } }
 
-      it { expect { subject }.to change(User, :count).by(1) }
+      context 'without a username in the request' do
+        let(:request) do
+          OpenStruct.new(env: { 'HTTP_X_FIRSTNAME' => 'foo', 'HTTP_X_LASTNAME' => 'bar',
+                                'HTTP_X_EMAIL' => 'foo@example.org' })
+        end
+
+        before do
+          allow(User).to receive(:find_nobody!)
+        end
+
+        it { is_expected.to be(true) }
+
+        it 'sets the current user to the anonymous user' do
+          subject
+          expect(User).to have_received(:find_nobody!)
```

Comment:
Testing with `User.session` doesn't work for the `nobody` user. So this is why I'm doing this.